lib/internal/async_local_storage/async_context_frame.js

```diff
@@ -4,10 +4,29 @@ const {
   ReflectApply,
 } = primordials;
 
+const {
+  validateObject,
+  validateString,
+} = require('internal/validators');
+
 const AsyncContextFrame = require('internal/async_context_frame');
 const { AsyncResource } = require('async_hooks');
 
 class AsyncLocalStorage {
+  #defaultValue = undefined;
+  #name = undefined;
+
+  constructor(options = {}) {
+    validateObject(options, 'options');
```

Comment:
I think we should only run this function if options is not equal to undefined. Otherwise we're making an unnecessary call for default case. This applies to other class as well.

Reply:
Given that the `options` arg has a default value it will never be `undefined`